### DISCUSSION TITLE
[search] Limit search to current location

### DISCUSSION THREAD

**Discussion by Shohreh** [organicmaps/organicmaps#6383] [Ideas]
Hello,

Currently, OM searches all the downloaded maps while looking for stuff.

Since people are more likely to be looking for something where they're located instead of another city (as displayed on the phone), I suggest making that the default, and provide the option to look in all the available maps.

Cheers,

![image](https://github.com/organicmaps/organicmaps/assets/1541861/8e6c5c18-ef1a-4864-8e94-cfe74b54906e)

**Comment by patepelo** [comment]
It is great to prioritize closer results but I wouldn't limit far away results to be non existent.

**Reply by Shohreh** [reply]
At least it should give the user the option to limit search, or even start by limiting itself to the zone currently displayed.

If I'm zooming in on a neighborhood in Madrid, I'm very unlikely to be looking for a place hundred's of kms away.

**Reply by patepelo** [reply]
Totally agree that its more probable to look for closer ones, but i've done long distance search before. It would be better maybe to make an "increase search radius" at the end of the results. I think OsmAnd does this.

**Reply by biodranik** [reply]
Those who are planning trips in OM do not need local search results.

**Comment by Shohreh** [comment]
While travelling, I very often need to look for stores and such _where I'm currently located_.

So I don't care about places hundreds of kms away.

I'm out.

**Reply by biodranik** [reply]
OM is used in different scenarios:
1. To find local features.
2. To find remote features (e.g. while planning a trip).

Dropping case (2) in favor of only case (1) is not a good UX.

**Reply by muralito** [reply]
I have seen another problem with the search and radius, probably caused by the fact that the distance in straight line is very much shorter than the distance by foot or car, like when there is a river/lake or a mountain/hill in the middle.
Could the search order be adjusted by calculating a more "real distance"? Maybe not by default (because it could be an expensive calculation),   but if the user ask for it...

I remember the problem here https://www.openstreetmap.org/#map=14/-31.3863/-57.9603 and here
https://www.openstreetmap.org/#map=15/-31.8239/-64.5235


About the using OM in different scenarios, for example,  the Overpass turbo wizard query has always worked well for me. Just using "in" or "around" in the query gives good results. Ej, "hotel in <city>", or "restaurant around <place>", but this wizard is in English language only, and OM needs to support several languages.
